internal/cmd/commands/server/worker_initial_upstreams_reload_test.go

```diff
@@ -91,25 +85,73 @@ func TestServer_ReloadInitialUpstreams(t *testing.T) {
 		t.Fatalf("timeout waiting for worker start")
 	}
 
-	// Wait until the worker has connected to the first controller
+	// Wait until the worker has connected to the first controller as seen via
+	// two status updates
 	timeout := time.NewTimer(15 * time.Second)
 	poll := time.NewTimer(0)
 	var w *server.Worker
+	var lastStatusTime time.Time
+	serversRepo, err := testController.Controller().ServersRepoFn()
+	require.NoError(err)
 pollFirstController:
 	for {
 		select {
 		case <-timeout.C:
 			t.Fatalf("timeout wait for worker to connect to first controller")
 		case <-poll.C:
-			serversRepo, err := testController.Controller().ServersRepoFn()
-			require.NoError(err)
 			w, err = serversRepo.LookupWorkerByName(testController.Context(), "test")
 			require.NoError(err)
 			if w != nil {
-				timeout.Stop()
-				break pollFirstController
+				switch {
+				case lastStatusTime.IsZero():
+					lastStatusTime = w.GetLastStatusTime().AsTime().Round(time.Second)
+					t.Log("recording initial last status time", lastStatusTime.Format(time.RFC3339))
+				default:
+					if !lastStatusTime.Equal(w.GetLastStatusTime().AsTime().Round(time.Second)) {
+						lastStatusTime = w.GetLastStatusTime().AsTime().Round(time.Second)
+						t.Log("recording updated last status time", lastStatusTime.Format(time.RFC3339Nano))
```

Comment:
Any reason with this one is using `RFC3339Nano` whereas the first log is using `RFC3339`?

Reply:
Ah, nope. I'll just remove them actually.